File: Backend/PF3311.TeleRehab.Api/Models/Therapy.cs
namespace PF3311.Telerehab.API.Models;

public class Therapy
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid PatientId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Instructions { get; set; } = string.Empty;
    public int Repetitions { get; set; }
    public string Frequency { get; set; } = string.Empty;
}

File: App2/telerehab_app/lib/models/therapy.dart
class Therapy {
  final String id;
  final String patientId;
  final String name;
  final String instructions;
  final int repetitions;
  final String frequency;

  Therapy({
    required this.id,
    required this.patientId,
    required this.name,
    required this.instructions,
    required this.repetitions,
    required this.frequency,
  });

  factory Therapy.fromJson(Map<String, dynamic> json) {
    return Therapy(
      id: json['id'] ?? '',
      patientId: json['patientId'] ?? '',
      name: json['name'] ?? '',
      instructions: json['instructions'] ?? '',
      repetitions: json['repetitions'] ?? 0,
      frequency: json['frequency'] ?? '',
    );
  }
}